scripts/bond-calculator.py

```diff
@@ -0,0 +1,148 @@
+    interest = get_interest_rate()
+    if options.exponent:
+        jm_single().config.set("POLICY", "bond_value_exponent", str(options.exponent))
+    if options.path_to_json:
+        try:
+            with open(options.path_to_json, "r", encoding="UTF-8") as orderbook:
+                bond_values = [fb["bond_value"] for fb in loads(orderbook.read())["fidelitybonds"]]
+            percentiles = quantiles(bond_values, n=100, method="inclusive")
+        except FileNotFoundError as exc:
+            log.error(exc)
+            sys.exit(EXIT_ARGERROR)
+        except StatisticsError:
+            log.error("Given JSON file does not contain enough fidelity bonds (at least 2 required)")
+            sys.exit(EXIT_ARGERROR)
+        bonds_sum = sum(bond_values)
+
+    current_time = datetime.now().timestamp()
+    try:
+        utxo_value = amount_to_sat(args[0])
+    except ValueError:
+        # If it's not a valid amount then it has to be a UTXO
+        if jm_single().bc_interface is None:
+            log.error("For calculation based on UTXO access to Bitcoin Core is required")
+            sys.exit(EXIT_FAILURE)
+        success, utxo = utxostr_to_utxo(args[0])
+        if not success:
+            # utxo contains the error message
+            log.error(utxo)
+            sys.exit(EXIT_ARGERROR)
+        utxo_data = jm_single().bc_interface.query_utxo_set(utxo, includeconf=True)[0]
+        utxo_value = utxo_data["value"]
+        current_height = jm_single().bc_interface.get_current_block_height()
+        block_hash = jm_single().bc_interface.get_block_hash(current_height - utxo_data["confirms"] + 1)
+        confirm_time = jm_single().bc_interface.get_block_time(block_hash)
+    else:
+        # We assume a fidelity bond with confirmation time equal to the current time.
+        # I.e., like if the fidelity bond UTXO with given amount has just confirmed on the blockchain.
+        confirm_time = current_time
+
+    jmprint(f"Amount locked: {utxo_value} ({sat_to_btc(utxo_value)} btc)")
+    jmprint(f"Confirmation time: {datetime.fromtimestamp(confirm_time)}")
+    jmprint(f"Interest rate: {interest} ({interest * 100}%)")
+    jmprint(f"Exponent: {jm_single().config.get('POLICY', 'bond_value_exponent')}")
+    jmprint("\nFIDELITY BOND VALUES")
+
+    locktime = add_one_month(
+        datetime.fromtimestamp(current_time).replace(day=1, hour=0, minute=0, second=0, microsecond=0))
+    for _ in range(options.months):
+        fb_value = FidelityBondMixin.calculate_timelocked_fidelity_bond_value(
+            utxo_value,
+            confirm_time,
+            datetime.timestamp(locktime),
+            current_time,
+            interest,
+        )
+        # Mimic the locktime value the user would have to insert to create such fidelity bond
+        jmprint(f"\nLocktime: {locktime.year}-{locktime.month}")
+        # Mimic orderbook value
+        jmprint(f"Bond value: {float(Decimal(fb_value) / Decimal(1e16)):.16f}")
```

Comment:
As in the orderbook, it probably makes sense to quote the unit (BTC $^{e}$, $e$=exponent) here.

Reply:
Done in 1637fbb7d7004c4f073bc34ec0e52c9d7a2d510f, I added it only once in the title, to avoid repetition, let me know if it's okay.